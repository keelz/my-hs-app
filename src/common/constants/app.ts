export default {
  COLLECTION: {
    CARDS_PER_PAGE: 10,
    FILTERS: {
      COST: 'COST',
      PLAY_STYLE: 'PLAY_STYLE',
      SET: 'SET',
    },
  },
};
