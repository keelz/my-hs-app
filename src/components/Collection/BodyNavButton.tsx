import * as React from 'react';
import { IComponentProps, BlockOrientation } from '../../common/models/app.model';
import { composeClassname } from '../../common/utils';

interface IBodyNavButtonProps extends IComponentProps {
  align: BlockOrientation.LEFT | BlockOrientation.RIGHT;
  onClick?: () => void;
}

export const composeAlignmentClassname = (align: BlockOrientation) =>
  align === BlockOrientation.LEFT
    ? 'Collection-align-left'
    : 'Collection-align-right';

export const composeIconContainerClassname = () =>
  'Collection-body-nav-button-icon';

export const composeIconClassname = (align: BlockOrientation) =>
  align === BlockOrientation.LEFT
    ? 'fas fa-angle-left'
    : 'fas fa-angle-right';

const BodyNavButton: React.SFC<IBodyNavButtonProps> = props =>
  <div
    className={composeClassname([
      'Collection-body-nav-button',
      composeAlignmentClassname(props.align),
    ])(props.className)}
    onClick={props.onClick}>
    <div className={composeIconContainerClassname()}>
      <i className={composeIconClassname(props.align)} />
    </div>
  </div>;

export default BodyNavButton;
