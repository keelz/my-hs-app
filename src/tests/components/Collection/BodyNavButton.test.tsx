import * as React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { BlockOrientation } from '../../../common/models/app.model';
import BodyNavButton, {
  composeAlignmentClassname,
  composeIconClassname,
  composeIconContainerClassname,
} from '../../../components/Collection/BodyNavButton';

const defaultProps = Object.freeze({
  active: false,
  align: BlockOrientation.LEFT,
  onClick: jest.fn(),
});

describe('Collect BodyNavButton', () => {
  describe('snapshots', () => {
    it('renders with orientation LEFT correctly', () => {
      const tree = enzyme.shallow(<BodyNavButton {...defaultProps} />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it('renders with orientation RIGHT correctly', () => {
      const props = { ...defaultProps };
      props.align = BlockOrientation.RIGHT;
      const tree = enzyme.shallow(<BodyNavButton {...props} />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });

  describe('integration', () => {
    it('handles onClick event correctly', () => {
      const tree = enzyme.mount(<BodyNavButton {...defaultProps} />);
      tree.simulate('click');
      expect(defaultProps.onClick).toHaveBeenCalled();
    });
  });

  describe('helpers', () => {
    it('composes alignment classnames correctly for LEFT alignment', () => {
      const test = composeAlignmentClassname(BlockOrientation.LEFT);
      expect(test).toMatchSnapshot();
    });

    it('composes alignment classnames correctly for RIGHT alignment', () => {
      const test = composeAlignmentClassname(BlockOrientation.RIGHT);
      expect(test).toMatchSnapshot();
    });

    it('composes icon container classname correctly when not active', () => {
      const test = composeIconContainerClassname({
        ...defaultProps,
        active: false,
      });
      expect(test).toMatchSnapshot();
    });

    it('composes icon container classname correctly when active', () => {
      const test = composeIconContainerClassname({
        ...defaultProps,
        active: true,
      });
      expect(test).toMatchSnapshot();
    });

    it('composes icon classname correctly for LEFT alignment', () => {
      const test = composeIconClassname(defaultProps);
      expect(test).toMatchSnapshot();
    });

    it('composes icon classname correctly for RIGHT alignment', () => {
      const props = { ...defaultProps, align: BlockOrientation.RIGHT };
      const test = composeIconClassname(props);
      expect(test).toMatchSnapshot();
    });
  });
});
