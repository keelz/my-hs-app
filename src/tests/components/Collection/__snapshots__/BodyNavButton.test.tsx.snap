// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collect BodyNavButton helpers composes alignment classnames correctly for LEFT alignment 1`] = `"Collection-align-left"`;

exports[`Collect BodyNavButton helpers composes alignment classnames correctly for RIGHT alignment 1`] = `"Collection-align-right"`;

exports[`Collect BodyNavButton helpers composes icon classname correctly for LEFT alignment 1`] = `"fas fa-angle-left"`;

exports[`Collect BodyNavButton helpers composes icon classname correctly for RIGHT alignment 1`] = `"fas fa-angle-right"`;

exports[`Collect BodyNavButton helpers composes icon container classname correctly when active 1`] = `"Collection-body-nav-button-icon active"`;

exports[`Collect BodyNavButton helpers composes icon container classname correctly when not active 1`] = `"Collection-body-nav-button-icon"`;

exports[`Collect BodyNavButton snapshots renders with orientation LEFT correctly 1`] = `
<div
  className="Collection-body-nav-button Collection-align-left"
  onClick={[MockFunction]}
>
  <div
    className="Collection-body-nav-button-icon"
  >
    <i
      className="fas fa-angle-left"
    />
  </div>
</div>
`;

exports[`Collect BodyNavButton snapshots renders with orientation RIGHT correctly 1`] = `
<div
  className="Collection-body-nav-button Collection-align-right"
  onClick={[MockFunction]}
>
  <div
    className="Collection-body-nav-button-icon"
  >
    <i
      className="fas fa-angle-right"
    />
  </div>
</div>
`;
